### DISCUSSION TITLE
Add the ability to include field values in UIFormatters format definition

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5873] [Feature Requests]
>My main issue is the generation of an accession number, because we cannot implement an automatic concatenation of other fields as we intended to do (the “index” field was thought for that purpose). But I suppose there is a way to automatically generate an accession number, i.e. the unique number that is attributed to an object as soon as it is registered in the collection (seems to be your concept also J). 
> 
>I’d like this accession number to be related to year of registration, level of collection, place and mode of conservation. Is that possible to have an automatically generated one (to avoid duplicate) providing these data?

**Requested By:** Rousse Pascal at [Chlordécone : les connaissances scientifiques](https://www.chlordecone-infos.fr/)

**Comment by maxpatiiuk** [comment]
How about allowing DataObjectFormatters inside of UIFormatters?
(because the other way around is already allowed)

**Comment by grantfitzsimmons** [comment]
> We have several examples in our collections where being able to search for and/or export a concatenation of fields would be very helpful (as well as displayed on a form). For the herbarium they store their unique preparation ID in three fields i.e. [prefix] [accession number]-[suffix] and many users would want to search for the combination. It would be great if we could configure a spare text field much in the same way that the formatted views for the queries are created - being able to string multiple fields and include any separators. For our purpose these "calculated" fields would be read-only, but would need to be available for query searching and use in labels and reports. Seeing the calculated field update in real-time on a form would be great, but having it update on a page refresh would also work.

**Requested By:** CSIRO on [Asana](https://app.asana.com/0/1201972579963460/1205419562512633/f)

**Comment by grantfitzsimmons** [comment]
This was discussed again by CSIRO today
